packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+// Internal Imports
+import { Component } from "../component";
+import { Events } from "../../interfaces";
+import { Tools } from "../../tools";
+import { DOMUtils } from "../../services";
+import { ChartModel } from "../../model";
+import * as Configuration from "../../configuration";
+
+// D3 Imports
+import { event, select } from "d3-selection";
+
+// import the settings for the css prefix
+import settings from "carbon-components/es/globals/js/settings";
+
+export class ToolBar extends Component {
+	type = "tool-bar";
+
+	isZoomBarEnabled: boolean;
+
+	// ul options list element
+	overflowMenuOptions;
+
+	// overflow menu icon rect to control background
+	overflowMenuIcon: any;
+
+	resetZoomMenuItemId = "resetZoomMenuItem";
+
+	// x, y coordinate of overflow menu
+	overflowMenuX = 0;
+	overflowMenuY = 0;
+
+	zoomService = this.services.zoom;
+	// overflow menu icon background padding
+	iconPadding = Configuration.toolBar.iconPadding;
```

Comment:
why would we store `iconPadding` on `this`? why can't we just grab it from `Configuration` each time?

Reply:
or can we store in a variable inside the method that needs it?

Reply:
anytime you're adding an attribute to the class it'll export it out to other places that have initialized an object instance of this class. It seems unnecessary to give access to the `zoomService` through an attribute

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/1e6ff2aca938073a1ddbd96fd979a27d72525026.